### PR TITLE
(admin/config only) midround slaughter demons

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Slaughter demons used to be part of a rare, late-round random event before dynamic was ported from /vg/.

Then this happened
https://github.com/tgstation/tgstation/blob/5670e238aa9d87c5416a157cf90be258aa4641c2/code/modules/events/ghost_role/slaughter_event.dm#L8
...and people slowly but surely forgot it was a thing.

So, slaughter demons have been brought back; tweaked a little so they don't roll when less than 30 (from 20) players are around, since they can easily pick players off one by one with hit and run tactics. Required round time reduced a smidge too, because dynamic runs on a budget that might get exhausted sooner or later. Coincidentally, that's also when heavy over light midround rulesets are guaranteed.

**EDIT: Because our esteemed (manuelites) headmins are not quite, understandably so, fond of the returnal of slaughter demons as rare midround antags, they'll be disabled by default, much like other rulesets such as meteors and clown ops. Though I think it's fair to keep them around, since their removal wasn't intended to begin with.**

## Why It's Good For The Game
Fixes a long-standing oversight that made the game a smidge less chaotic.

## Changelog

:cl:
admin: Locked behind badminnery and configs, but slaughter demons as midround antags are back.
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
the slaughter demon sweep in the screenshot trsts is hilarious. would vote next president

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -925,5 +925,60 @@
...
 
+/// Midround Slaughter Demon Ruleset (From Ghosts)
+/datum/dynamic_ruleset/midround/from_ghosts/slaughter_demon
+	name = "Stray Slaugther Demon"
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_HEAVY
+	antag_datum = /datum/antagonist/slaughter
+	antag_flag = ROLE_SLAUGHTER_DEMON
+	required_enemies = list(4, 4, 4, 3, 3, 2, 2, 1, 0, 0)
+	required_candidates = 1
+	weight = 1 //very rare
```
[the remark this answers is not in the document]
Comment (Ghommie) [reply code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:936]: So, should I bump it up or down?

**Comment by nukularpower** [comment]
Does this include laughter demons?  If not, pretty please make it so :D

**Comment by Ghommie** [comment]
> Does this include laughter demons?

It does.

**Comment by optimumtact** [comment]
headadmins said they didn't want it and i don't see any reason to override them.

So this will be closed in 24 hours.

**Comment by optimumtact** [comment]
note you might be able to rescue this PR's code if you make them default to never being picked so they canb e configured on by a future term, but im' not dynamic expert.

**Comment by Ghommie** [comment]
Easy as changing the requirements to 101.

**Comment by IndieanaJones** [comment]
You could just remove the dynamic_should_hyjack line and revert it to the original behavior of insanely rare event spawn.

**Comment by Ghommie** [comment]
> You could just remove the dynamic_should_hyjack line and revert it to the original behavior of insanely rare event spawn.

Antags picked by the game should generally cost threat points, and I don't think having a slaughter demon spawn regardless of overall threat, spent nor unspent, is a good idea.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I don't want this in the codebase even off by default because we don't have any way to hide the preference icon even if the ruleset is completely unused, which would just create clutter. I think we should instead just save this for when head admins want it because it's not much code at all to add

**Pull request closed without merging**